apps/api-extractor/src/analyzer/PackageMetadataManager.ts

```diff
@@ -93,8 +93,19 @@ export class PackageMetadataManager {
       // 3. If package.json contains a field such as "main": "./path1/path2/index.js", then we look for
       // the file under "./path1/path2/tsdoc-metadata.json"
       tsdocMetadataRelativePath = path.join(path.dirname(packageJson.main), tsdocMetadataFilename);
+    } else if (
```

Comment:
Spec correct ordering in TypeScript is:
1. exports (which has `types` and the ability to do `types@{selector}` where `{selector}` is an expression compatible with `typesVersions`
2. typesVersions
3. types/typings
4. main with extension substitution

The relative ordering of `exports` is not clearly documented in TypeScript, but in NodeJS it takes precedence over `main`

Reply:
I'm thinking we should default to matching the newest version of a versioned types field. @octogonz - do you agree?

Reply:
Yes.  After some discussion with @iclanton, this lookup needs to be deterministic and should not depend on the compiler version.

In other words, if a given API entry point has paths for multiple compiler versions, then it is reasonable to assume that either all compiler versions support AEDoc, or none of them do.  And so it is reasonable for the `<lookup>` heuristic to choose the latest version.